pyvista/core/filters.py

```diff
@@ -1129,6 +1129,24 @@ def warp_by_scalar(dataset, scalars=None, factor=1.0, normal=None,
             return
         return output
 
+    def warp_by_vector(dataset, scalars=None, factor=1.0):
+        """Warp the dataset's points by a point data vectors array's values.
+
+        This modifies point coordinates by moving points along point vectors by
+        the local vector times the scale factor.
+
+        A classical application of this transform is to visualize eigenmodes in mechanics.
+        """
+        if scalars is None:
+            field, scalars = dataset.active_scalars_info
+        arr, field = get_array(dataset, scalars, preference='point', info=True)
+        alg = vtk.vtkWarpVector()
+        alg.SetInputDataObject(dataset)
+        alg.SetInputArrayToProcess(0, 0, 0, field, scalars)
+        alg.SetScaleFactor(factor)
+        alg.Update()
+        output = _get_output(alg)
+        return output
```

Comment:
Consider updating "in-place".  You won't have to return an entirely new mesh.  Look at other methods/functions that use the `in_place` keyword argument for an example.